### PR TITLE
Helper functions to stream data from HDF5

### PR DESCRIPTION
**Pull request by WardLT**
So that we can analyze files larger than what fits in memory.

### CODE REVIEW

File: batdata/streaming.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def iterate_cycles_from_file(hdf5_path: Union[Path, str, HDFStore],
+                             make_dataset: bool = False,
+                             **kwargs) -> Iterator[Union[pd.DataFrame, BatteryDataset]]:
+    """Stream single-cycle datasets from the HDF5 file
+
+    Keyword arguments are passed to :meth:`~pandas.read_hdf`.
+
+    Args:
```
Comment (victorventuri) [inline batdata/streaming.py:49]: What are the `**kwargs` used for? I noticed in the tests that you had a `chunksize` as one of them?

**Review by victorventuri — COMMENTED** [review]
I think this looks great! I can understand how the `iterate_records` work by loading data row by row, but I am unclear on what the cycle is doing with the chunks. Is it loading chunk by chunk while still in the same cycle until it is no longer in the same cycle? Is that the idea behind chunking it, so you don't have to load in a lot of data and then `groupby('cycle_number')`? 

I think this (along with the previous one) would both benefit from a simple notebook example: load a cycle, perform estimations, save to disk.

File: batdata/streaming.py (hunk 1 above)
Comment (WardLT) [reply batdata/streaming.py:49]: They control how the HDF5 reader backing pandas works. Options include changing which columns are read, a query string, etc.

**Comment by WardLT** [comment]
I'll add some docs about the "iterate over cycles." We describe them as "groups of cycles," which is not quite right (we return a single cycle per iterator). I'm going to hold off on a notebook for now, since I'd prefer the documentation to be sufficient rather than having content be spread out in two places (docs, notebooks).

**Pull request merged**